webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/EndpointsDefinition.scala

```diff
@@ -271,25 +272,25 @@ object GroupApi       extends Enum[GroupApi] with ApiModuleProvider[GroupApi] {
     val z: Int = implicitly[Line].value
     val description    = "Get information about the given group category"
     val (action, path) = GET / "groups" / "categories" / "{id}"
-    override def dataContainer: Some[String] = Some("groupCategories")
+    override def dataContainer: None.type = None
```

Comment:
isn't that changing the resulting JSON ? I think it will lead to have one less `{ ..., "groupCategories": [ ... content ...] }`

If so, you can't do that. We don't have the right to change external APIs without a fallback path for their consumer. Typically, if a script consume it for extracting info about the category, it will break, and it can't be repaired by using previous API version in that case. If it changes nothing in the output, you can ignore that message :)

Reply:
That is a caveat of the migration : the `dataContainer` is used in the plumbings of the API only to return an ARRAY within the data : `{...,data: { groupCategories : [] }}`, and using `toLiftResponseOne` will make an array with a single element...
However here there is an existing test which shows that we have `{...,data: { groupCategories : {} }}` with an object instead, so we have to suppress the `dataContainer` and wrap the Json within an object under a `groupCategories` field (a wrapper `case class Wrapper(groupCategories: GroupCategory)`) :
![image](https://github.com/user-attachments/assets/36062239-3016-4d15-ba5d-7060fe252252)

Reply:
(the case you addressed was covered by this test, which has not changed)